Add support for backing up disks during test (#2341)
os-autoinst/os-autoinst: doc/memorydumps.asciidoc

== How to use

Three methods are available to the test writer, `freeze_vm`, `save_memory_dump` and `save_storage`

The feature is enabled by calling the aforementioned methods, using

> a https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc#how-to-write-tests[post_fail_hook] and setting the test's flags as fatal. The `save_memory_dump`  and `save_storage` method's can be however

Reviewer: s/method's/methods/

NOTE: Call this method to ensure memory and disk dump refer to the same machine state.

=== save_storage

> If it is supported by the backend `save_storage` will save all storage disks into backup.

Reviewer: s/disks/devices/

> This can be called anywhere within test. For multiple calls within test module is needed to
> specify different filenames. By default it uses test module name.
> 
> NOTE: this method is space and time expensive.

Reviewer 2: ```suggestion
This can be called from anywhere within a test. For multiple calls within a test module a filename needs to be specified. By default it uses the test module name.

NOTE: This method is time-consuming and requires a lot of space.
```
